boxoffice/models/item.py

```diff
@@ -79,12 +92,79 @@ def quantity_available(self):
 
     @property
     def is_available(self):
-        """Checks if an item has a current price object and has a positive quantity_available"""
+        """Check if an item has a current price object and has a positive quantity_available"""
         return bool(self.current_price() and self.quantity_available > 0)
 
     def is_cancellable(self):
         return datetime.utcnow() < self.cancellable_until if self.cancellable_until else True
 
+    def url_for(self, action='view', _external=False, **kwargs):
+        if action == 'view':
+            return url_for('admin_item', item_id=self.id, _external=_external, **kwargs)
+
+    @property
+    def url_for_view(self):
+        return self.url_for('view')
+
+    @property
+    def active_price(self):
+        return self.current_price().amount if self.current_price() else None
+
+    @property
+    def get_confirmed_line_items(self):
+        """Returns a SQLAlchemy query object preset with an item's confirmed line items"""
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return LineItem.query.filter(LineItem.item == self, LineItem.status == LINE_ITEM_STATUS.CONFIRMED)
+
+    @property
+    def sold(self):
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return LineItem.query.filter(LineItem.item == self, LineItem.final_amount > 0, LineItem.status == LINE_ITEM_STATUS.CONFIRMED).count()
+
+    @property
+    def free(self):
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return LineItem.query.filter(LineItem.item == self, LineItem.final_amount == 0, LineItem.status == LINE_ITEM_STATUS.CONFIRMED).count()
+
+    @property
+    def cancelled(self):
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return LineItem.query.filter(LineItem.item == self, LineItem.status == LINE_ITEM_STATUS.CANCELLED).count()
+
+    @property
+    def net_sales(self):
+        from ..models import LineItem, LINE_ITEM_STATUS
+
+        return db.session.query(func.sum(LineItem.final_amount)).filter(LineItem.item == self,
+            LineItem.status == LINE_ITEM_STATUS.CONFIRMED).first()[0]
```

Comment:
Doesn't `first()` return a single instance and not a list?

Reply:
Also, same as discount policy. If `Item` had `line_items` as relationship, these 5 methods above will become much simpler. But this works for now.

Reply:
> Doesn't first() return a single instance and not a list?

No, since the query is on `db.session` and not on a specific model, it returns a tuple.

Reply:
Good point, I've added the lazy loader on the relationship and the queries are cleaner as a result.